cmd/logcli/main.go

```diff
@@ -193,8 +260,12 @@ func main() {
 
 		if *tail || *follow {
 			rangeQuery.TailQuery(time.Duration(*delayFor)*time.Second, queryClient, out)
-		} else {
+		} else if rangeQuery.ParallelMaxWorkers == 1 {
 			rangeQuery.DoQuery(queryClient, out, *statistics)
+		} else {
+			// `--limit` doesn't make sense when using parallelism.
+			rangeQuery.Limit = math.MaxInt
```

Comment:
```suggestion
			rangeQuery.Limit = 0
```

Checkout the changes in https://github.com/grafana/loki/pull/8537